java/query/src/main/java/sleeper/query/tracker/DynamoDBQueryTrackerEntry.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.query.tracker;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeAction;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.AttributeValueUpdate;
+
+import sleeper.query.model.LeafPartitionQuery;
+import sleeper.query.model.Query;
+import sleeper.query.model.output.ResultsOutputInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static sleeper.query.tracker.DynamoDBQueryTracker.NON_NESTED_QUERY_PLACEHOLDER;
+
+public class DynamoDBQueryTrackerEntry {
```

Comment:
We could rename this class to `DynamoDBQueryTrackerFormat` to match classes similar to this one which construct DynamoDB records (like `DynamoDBFileInfoFormat`)

Reply:
This class is a model for an individual entry in DynamoDB, where DynamoDBFileInfoFormat is a format which can convert between another model and a DynamoDB item.

They both construct and read DynamoDB records but I think there's a meaningful difference between modelling an entry and modelling the format in abstract?

Reply:
Ahh I see now, that makes sense